pkg/utils/miniojob/types.go

```diff
@@ -148,11 +141,48 @@ func (jobCommand *MinIOIntervalJobCommand) CreateJob(ctx context.Context, k8sCli
 		}
 	}
 	jobCommands = append(jobCommands, "--insecure")
-	objs := []client.Object{}
 	mcImage := jobCR.Spec.MCImage
 	if mcImage == "" {
 		mcImage = DefaultMCImage
 	}
+	baseVolumeMounts := []corev1.VolumeMount{
+		{
+			Name:      "config-dir",
+			MountPath: "/.mc",
+		},
+	}
+	baseVolumeMounts = append(baseVolumeMounts, jobCommand.CommandSpec.VolumeMounts...)
+	baseVolumes := []corev1.Volume{
+		{
+			Name: "config-dir",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		},
+	}
+	baseVolumes = append(baseVolumes, jobCommand.CommandSpec.Volumes...)
+	baseEnvFrom := []corev1.EnvFromSource{
+		{
+			SecretRef: &corev1.SecretEnvSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: fmt.Sprintf("%s-job-secret", jobCR.Name),
+				},
+			},
+		},
+	}
+	baseEnvFrom = append(baseEnvFrom, jobCommand.CommandSpec.EnvFrom...)
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-job-secret", jobCR.Name),
+			Namespace: jobCR.Namespace,
+		},
+		StringData: map[string]string{
+			"MC_HOST_myminio":                    fmt.Sprintf("https://$(ACCESS_KEY):$(SECRET_KEY)@minio.%s.svc.cluster.local", jobCR.Namespace),
+			"MC_STS_ENDPOINT_myminio":            fmt.Sprintf("https://sts.%s.svc.cluster.local:4223/sts/%s", miniov2.GetNSFromFile(), jobCR.Namespace),
```

Comment:
port is always 4223?

Reply:
https://github.com/minio/operator/blob/2301b554e1076ab0ee8ba266130f191d5673b565/pkg/controller/sts.go#L37
```golang
	STSDefaultPort = "4223"
```
@shtripat